compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp

```diff
@@ -1114,6 +1114,86 @@ void propagate_pack_quantparam(luci::CirclePack *pack, loco::DataType quant_type
   }
 }
 
+/** EXAMPLE
+ *
+ *
+ *
+ * BEFORE
+ *
+ *      [CircleNode] [CircleConst] [CircleConst] [CircleNode]
+ *          (S32)        (S32)        (FP32)     (U8 qparam1)
+ *              \          \           /            /
+ *               \          \        /            /
+ *                \          \     /            /
+ *                 -------[CircleOneHot]-------
+ *                         (U8 qparam2)
+ *
+ *  AFTER
+ *
+ *      [CircleNode] [CircleConst] [CircleConst] [CircleNode]      [CircleConst] <- Dead node
+ *          (S32)        (S32)     (U8 qparam2)  (U8 qparam2)         (FP32)
+ *              \          \           /           /
+ *               \          \        /            /
+ *                \          \     /            /
+ *                 -------[CircleOneHot]-------
+ *                         (U8 qparam2)
+ *
+ * NOTE Quantization parameter of CircleOneHot (qparam2) is propagated to on_value/off_value.
+ */
+void propagate_one_hot_quantparam(luci::CircleOneHot *one_hot, loco::DataType quant_type)
+{
+  assert(one_hot->quantparam() != nullptr);
+
+  // Propagate quantization paramters from output to inputs,
+  // to fit both input and counstant_value in one quant range.
+  auto quant_input = [one_hot, quant_type](void (CircleOneHot::*arg_setter)(loco::Node *),
+                                           loco::Node *(CircleOneHot::*arg_getter)() const) {
+    auto node = loco::must_cast<luci::CircleNode *>((one_hot->*arg_getter)());
+
+    // Quantize constant values
+    if (node->opcode() == luci::CircleOpcode::CIRCLECONST)
+    {
+      luci::CircleConst *const_node = loco::must_cast<luci::CircleConst *>(node);
+      if (is_quantized(const_node))
+        return;
+
+      if (const_node->dtype() != loco::DataType::FLOAT32)
+        throw std::runtime_error("Unsupported data type for constant input of OneHot Op");
```

Comment:
@m-bronnikov , @jinevening , Q) can there be other types than F32 as normal type?
if this(`const_node->dtype() == loco::DataType::FLOAT32`) is a MUST then throwing seems OK but if there can be other types, then it would be better to just return

Reply:
@seanshpark Please note this lambda is a bit modififed version of lambda from `propagate_pad_v2_quantparam`, which contains following lines:

https://github.com/Samsung/ONE/blob/0fb144c4d75526b7ffe6e2befa11905dab236659/compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp#L1433-L1434

Also, this lambda calls only for `on_value` and `off_value` nodes, which must have same type, as output node, which must have qparam, and, therefore `on_value`/`off_value` must has FP32 type or be already quantized. If node quantized, this lambda makes `return` (see above).

Reply:
Thanks for the explanation!